.github/workflows/artifacts.yml

```diff
@@ -0,0 +1,245 @@
+            echo "PARTNER_CHAINS_CLI_AARCH64_APPLE_DARWIN=partner-chains-cli-${{ github.event.inputs.partner-chains-tag }}-aarch64-apple-darwin" >> $GITHUB_ENV
+            echo "PARTNER_CHAINS_NODE_AARCH64_APPLE_DARWIN=partner-chains-node-${{ github.event.inputs.partner-chains-tag }}-aarch64-apple-darwin" >> $GITHUB_ENV
+          else
+            echo "PARTNER_CHAINS_CLI_AARCH64_APPLE_DARWIN=partner-chains-cli-aarch64-apple-darwin" >> $GITHUB_ENV
+            echo "PARTNER_CHAINS_NODE_AARCH64_APPLE_DARWIN=partner-chains-node-aarch64-apple-darwin" >> $GITHUB_ENV
+          fi
+
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.inputs.partner_chains_sha }}
+
+      - name: Install protoc
+        run: |
+          curl -LO https://github.com/protocolbuffers/protobuf/releases/download/v21.3/protoc-21.3-osx-aarch_64.zip
+          unzip protoc-21.3-osx-aarch_64.zip -d $HOME/protoc
+          sudo mv $HOME/protoc/bin/protoc /usr/local/bin/protoc
+
+      - name: Install Rust toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          default: true
+          profile: minimal
+
+      - name: Build partner-chains-node
+        run: |
+          rustup target add aarch64-apple-darwin
+          cargo build -p partner-chains-node --locked --release --target aarch64-apple-darwin
+          cp target/aarch64-apple-darwin/release/partner-chains-node $PARTNER_CHAINS_NODE_AARCH64_APPLE_DARWIN
+          chmod +x $PARTNER_CHAINS_NODE_AARCH64_APPLE_DARWIN
+
+      - name: Build partner-chains-cli
+        run: |
+          rustup target add aarch64-apple-darwin
+          cargo build -p partner-chains-cli --locked --release --target aarch64-apple-darwin
+          cp target/aarch64-apple-darwin/release/partner-chains-cli $PARTNER_CHAINS_CLI_AARCH64_APPLE_DARWIN
+          chmod +x $PARTNER_CHAINS_CLI_AARCH64_APPLE_DARWIN
+
+      - name: Upload partner-chains-cli-aarch64-apple-darwin
+        uses: actions/upload-artifact@v4
+        with:
+          name: ${{ env.PARTNER_CHAINS_CLI_AARCH64_APPLE_DARWIN }}
+          path: ${{ env.PARTNER_CHAINS_CLI_AARCH64_APPLE_DARWIN }}
+            
+      - name: Upload partner-chains-node-aarch64-apple-darwin
+        uses: actions/upload-artifact@v4
+        with:
+          name: ${{ env.PARTNER_CHAINS_NODE_AARCH64_APPLE_DARWIN }}
+          path: ${{ env.PARTNER_CHAINS_NODE_AARCH64_APPLE_DARWIN }}
+
+  partner-chains-smart-contracts-build:
+    runs-on: [self-hosted, nixos]
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          repository: input-output-hk/partner-chains-smart-contracts
+          token: ${{ secrets.ACTIONS_PAT }}
+          ref: ${{ github.event.inputs.partner_chains_smart_contracts_sha }}
+          path: partner-chains-smart-contracts
+
+      - name: Build
+        run: nix build ./partner-chains-smart-contracts#sidechain-release-bundle
```

Comment:
Oh, please no. We should just download artifact from the other repo at most. Why do we build it here?

Reply:
I agree, removed